### ISSUE TITLE
React Native iOS Deploy build failing

### ISSUE THREAD

**Issue by ajonno** [bitrise-io/build.issues#19]
# Bitrise Build Issue Report template

## Description of the issue

Tried to run a "Deploy" branch to test out Bitrise (for app distribution) on a React Native iOS app. The build crashed. Just to note our local builds are fine / can generate an IPA.

This was the error:
/Users/vagrant/git/ios/cleverbean/AppDelegate.m:14:9: 'RCTBundleURLProvider.h' file not found
#import "RCTBundleURLProvider.h" 

## Environment:

This was run on the "Deploy" branch of Bitrise.

_If on Bitrise.io: which stack? If not on Bitrise.io: on what operating system? (Plus any other information you can share)_

The Bitrise build stack used was Xcode 8.1.x, on macOS 10.12 (Sierra).  This matches our local dev setup.

__Which build Step causes the issue and which version of the step?__

Here is the relevant Bitrise log error report:
Running script '[CP] Check Pods Manifest.lock'
▸ Compiling main.m
▸ Compiling AppDelegate.m

❌  /Users/vagrant/git/ios/cleverbean/AppDelegate.m:14:9: 'RCTBundleURLProvider.h' file not found
#import "RCTBundleURLProvider.h"
        ^
▸ Compiling cleverbean_vers.c
▸ Compiling AppDelegate.m
** ARCHIVE FAILED **

## Reproducibility

- _Does a "Rebuild" help? (You can trigger a rebuild from the Build's page, by clicking the "Rebuild" button in the top right corner of a finished build)_ : 
Tried this --> NO

- _Does a rebuild without caches help? (You can remove the `Cache:Pull` and `Cache:Push` steps temporarily to not to use the cache, or [you can *delete all the caches* on the `Settings` tab of the app](https://bitrise-io.github.io/devcenter/caching/about-caching/#downloading-and-deleting-caches)._ :    
NO. i went to the Settings tab of the app, but there was NO option to disable the "Manage Build Caches (BETA)".. ??

- _If it's an issue which happens sporadically, what's the frequency? (e.g. Once a day ; about x% of the builds)_ :
HAPPENS ON EVERY BUILD

- _Does upgrading the build Step to the latest version help?_ : YES/NO
NOT SURE 

- _When did the issue start?_ :
YESTERDAY

### Linux/Android stack builds

_Can it be reproduced by running the build locally, **after doing a clean git clone (git clone the repository, into a new directory, somewhere on your Mac/PC)** and running the build from there with the Bitrise CLI ( https://www.bitrise.io/cli )? 
LOCAL BUILD ARE FINE. I CAN CREATE AN IPA EVEN AFTER GIT CLONE TO A NEW DIR.

## Build log

[c2810b27c28940a4.pdf](https://github.com/bitrise-io/build.issues/files/632705/c2810b27c28940a4.pdf)

_Please attach the build log (you can download the build log from the build's page,
once the build is finished, using the "Download log" button - floating at the bottom right corner of the log viewer),

DONE
[raw-xcodebuild-output.pdf](https://github.com/bitrise-io/build.issues/files/632709/raw-xcodebuild-output.pdf)

**Comment by viktorbenei** [comment]
I believe you forgot to run `npm install`, that's why the header is missing. See https://github.com/brentvatne/react-native-svgkit/issues/11#issuecomment-169336433

**Comment by ajonno** [comment]
still failing. i inserted npm install via a "do anything" step. this is error:

Compiling RCTFBSDKShareDialog.m

❌  /Users/vagrant/git/node_modules/react-native-fbsdk/ios/RCTFBSDK/share/RCTFBSDKShareDialog.h:21:9: 'FBSDKShareKit/FBSDKShareKit.h' file not found

#import <FBSDKShareKit/FBSDKShareKit.h>
        ^

**Comment by viktorbenei** [comment]
That's a separate issue, related to Facebook React SDK - something like this should help:

* http://stackoverflow.com/questions/35590903/xcode-always-says-fbsdksharekit-fbsdksharekit-h-file-not-found
* or: http://stackoverflow.com/questions/30136279/fbsdkcorekit-fbsdkcorekit-h-not-found-error

**Comment by ajonno** [comment]
i hear you - although nothing ive tried so far is working. just need someone with experience fixing this problem, re the correct way and place to enter a 'path' that the CI build can see.  why is it that the Primary channel is building ok, but the Deploy channel is not ?

**Comment by viktorbenei** [comment]
> why is it that the Primary channel is building ok, but the Deploy channel is not ?

Without the code it's impossible to say. Might be a configuration issue e.g. different Xcode schemes/configuration, where one is configured correctly but not the other. Or that the SDK is only used in one config but not the other. There's absolutely 0 difference from bitrise perspective, every build runs the exact same way, executing the build configuration you specify with the source code you supply.

> re the correct way and place to enter a 'path' that the CI build can see

Any path can work, just don't use absolute paths. Usually `$HOME/something` or `~/something` is a good idea if you have to download resources into a central place, as `$HOME` (or `~`) will always exist on every machine, but e.g. `/Users/MyUser` won't. Relative paths should also work, the only thing you should avoid is using / hardcoding absolute paths (like `Users/MyUser/something/...`), as that will only work on a specific machine, where that specific user/path exists.

If possible you should try to build the project on a Mac where you never built it before, you'll face the same issue there. The best environment is a virtual machine for this, but creating a macOS virtual machine might not be a simple task. We have a tool in alpha which can help you to create a VirtualBox macOS virtual machine, if you're interested: https://github.com/bitrise-tools/replica

**Comment by viktorbenei** [comment]
I'll close this issue, but if you need more info/help @ajonno just let us know and we'll reopen it!
